File: backend/src/main/java/com/slior/dto/auth/AuthResponse.java
package com.slior.dto.auth;

import com.slior.model.enums.UserRole;

import java.util.UUID;

/**
 * DTO de salida para los endpoints de autenticación.
 * Contiene el JWT y datos básicos del usuario.
 * NUNCA incluye el password.
 */
public record AuthResponse(

        String token,

        String type,       // Siempre "Bearer"

        UUID userId,

        String nombre,

        String email,

        UserRole rol
) {}

File: mobile-app/app/src/main/java/com/slior/data/remote/dto/AuthResponse.kt
package com.slior.data.remote.dto

/** Respuesta del servidor al hacer login o register. Contiene el JWT. */
data class AuthResponse(
    val token: String,
    val type: String,
    val userId: String,
    val nombre: String,
    val email: String,
    val rol: String
)